dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java

```diff
@@ -184,6 +188,38 @@ public Response getMessages(@Context HttpServletRequest request,
         return oldLanguagesResource.getMessages(request, i18NForm);
     }
 
+    /**
+     * Gets all the Messages from the language passed.
+     *
+     * If no language is passed will use the default language.
+     * @param request
+     * @param response
+     * @param language languageId or languageCountryCode e.g en_us, it_it
+     * @return all the messages of the language
+     */
+    @GET
+    @Path("/i18n/all")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response getAllMessages (
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            @DefaultValue("-1") @QueryParam("language") final String language){
```

Comment:
"language" shouldn't be a query parameter. 
Pathing should be
`v2/languages/{language}/i18n` to get all the messages for a language

Reply:
Requesting changes above

Reply:
The lang is optional, that's why is a queryparam.

Reply:
done